GPX language conversions and code cleanup

andrixnet opened the pull request:
While trying to fix https://github.com/opencaching/opencaching-pl/issues/917 I encountered several inconsistencies and i tried to resolve them as best I could. 

Most search.*.inc.php defined the same functions to do some work for them. I moved those functions to common.inc.php with several improvements. 

myroutes_search.php uses the same function names, but slightly different implementation. For now I renamed them to isolate them. Can this be merged somehow with the search.*.inc.php borne functions? 
A similar situation is in ocxml11.php. I know this is a sensitive API to older apps, so I took the same action as for myroutes_search and ask the same question. 

Is the OVL format used ? 
I found a syntax error in search.ovl.inc.php that was left there probably since DB refactoring over half an year ago. 

I kept Plconvert (renamed as conver_lang) but didn't use it. 
I found some code that makes use of iconv, which I consider a much better choice then a manual replacement list (which would work fine for a single language). 
**I replaced the use of convert_lang with convert_text (uses iconv)**
This must be tested in other languages, I only tested with romanian content. Seems fine. 

I made some changes to loadlanguages with setlocale (ro was missing, added upcoming hu and added LC_CTYPE, recommended for iconv usage).

---
Please review my changes and test them in your development machine with other language data then english and romanian.

Thank you. 

P.S. 
future note to self: these file exports redefine a whole lot of data definitions (like cache types, cache sizes, etc), so they will have to be heavily reworked for https://github.com/opencaching/opencaching-pl/projects/2

kojoty commented on lib/js/GCT.lang.php:
```diff
@@ -0,0 +1,22 @@
+
+header("Content-Type: text/javascript");
+
+echo("
```
I think much more readable and easier in maintenance is to close php in such situations for example here you can (probably) just do:

_(...)
?>

GCT_lang = {
  prev: <?=tr('pagination_left')?>,
  next: <?=tr('pagination_right')?>,
  number_of_caches: <?=tr('number_of_caches')?>,
 }_

mzylowski replied:
@kojoty Your sugestion from here is solved by andrie via #928

kojoty commented:
Hi Andrei,

Generally this is a good idea, but I have a few opinion about this:
1. In my opinion we should avoid adding anything to the common.inc.php - actually I would like one day to remove this file at all but there is still too much important code - of code is needed by some subset of scripts it should be moved to separate library and included by this subset of scripts - not by all of the scripts - so in this case you can consider to move this functions to the file called for example search.common.inc.php and include this file wherever it is needed. Alternatively if I remember well all search*.inc files are loaded only by search.php so maybe you can put this functions to the search.php - then they should be accessible from all search.*.inc.php scripts

2. Do you know why we need to convert from utf-8 to ascii for gpx files? - I don't know much about gpx and maybe this is reasonable but I also would not be surprised if this conversion is not necessary at all. Do you have some knowledge about it?

another comments in the code...

mzylowski commented:
I will try test your change on my devel in tomorrow evening

andrixnet commented:
@kojoty 
I moved those to common.inc.php for the reason that they were used in lib.search.*.inc.php, myroutes_search.php and ocxml11.php (which still needs clarification). Not only as included from search.php
Of course, it can be another library file. another library can be though of. For example there are many GPX definitions that are redefined in each place they are used, as @harrieklomp pointed out to me. 

Converting from UTF-8 to ASCII already existed in the code (though not working properly). 
Reasons for which to do this are third party applications and devices that can import GPX files but cannot handle or display UTF-8 characters properly or they simply don't have fonts with sufficient characters. One such example is GSAK. 

Since everything is UTF-8 in the website, and I assume everything else in the site works properly, then the data is consistent. 
I would not consider GPSBABEL for such a conversion, because it will lose fields deemed unnecessary by it. 

I would consider however a variant to offer GPX with UTF-8 content and converted GPX because after export is quite hard for the average end-user to make such conversions. 
(though I would also suggest filtering the resulting GPX through iconv utility on the server rather then dump the conversion on the user)

mzylowski commented:
@andrixnet After merging #928 here will be a conflict. To avoid, please remove cdf4446 from this PR.

About the rest: LGTM, except kojoty comments.
@kojoty Merge button in your hands.

This should be merged after #928, so for now 'Don't commit label'.

@andrixnet thanks for this PR ;]

andrixnet commented:
Keep Don't commit label. I have further improvements on the way.

andrixnet commented on lib/languages/pl.php:
```diff
@@ -1118,6 +1118,8 @@
     'route_03' => 'Wyczyść trasę',
     'route_04' => 'Reset mapy',
     'route_05' => 'Wczytuję trasę ...',
+    'route_06' => 'Maximalna ilość logów musi być cyfrą!',
```
Please confirm this translation. 
Also please confirm the meaning of the string. 

(old meaning said "minimum" in a context about a "maximum"

kojoty replied:
@andrixnet: better polish translation could be probably:
'route_06' => 'Maksymalna ilość logów musi być liczbą!',
'route_07' => 'Dozwolona wartość maksymalnej ilości logów musi być z zakresu: 0 - 999',
(you can change it or I'll fix it later)

andrixnet replied:
done

andrixnet commented:
@kojoty @mzylowski  @harrieklomp 
Please take a look at this. 
I've just finished working the file exports. 

Sorry for the one huge commit, I worked for many days for short periods between other work and I couldn't manage to assemble changes atomically. 
For a log of the changes please look mainly at https://github.com/opencaching/opencaching-pl/pull/931/commits/da93bd9e3201a64f36f6bd866362157f4f3466b2

Main improvements: 
- OC generated GPX files now are proper, albeit converted to ascii. Fields are consistent. Attributes are properly set. String language is consistent. ZIP is disabled (as it was set before)
- KML now has improved icons and consistent data
- opencaching.kml live Google Earth exporter is now generated using your profile's home coords or site home coords.
- Google Earth exporter uses same styles as KML file export. 
- KML styles based on c:geo marker_oc graphics
- WPT, TXT, UAM, OVL, OVM, LOC files improvements. 
- removed unused files
- removed unused bits of code.

A few translation changes and a few hardcoded strings translated. Please check.

andrixnet commented:
If there are no objections, may it be merged after this weekend?

kojoty commented:
@andrixnet, From my side - I'l try to review it tomorrow.

andrixnet commented on lib/js/GCT.lang.php:
```diff
@@ -0,0 +1,22 @@
+};
+");
+
+?>
```
well, removed it :-)

andrixnet commented on lib/search.ovl.inc.php:
```diff
@@ -155,7 +155,7 @@
 
         $nr = 1;
-        $s = dbcSearch->simpleQuery(
+        $s = $dbcSearch->simpleQuery(
```
Either it wasn't used for many months, or the few users that tried it (and failed) didn't bother to report it...

andrixnet replied:
I know nothing about this format or what devices make use of it.

andrixnet commented on xml/ocxml11.php:
```diff
@@ -365,7 +365,7 @@ function outputXmlFile($sessionid, $filenr, $bXmlDecl, $bOcXmlTag, $bDocType, $z
             FROM `cache_attrib` WHERE `language`='pl'");
         fwrite($f, $t1 . '<attrlist>' . "\n");
         while ($r = XDb::xFetchArray($rs)) {
-            fwrite($f, $t2 . '<attr id="' . $r['id'] . '" icon_large="' . xmlentities($absolute_server_URI . $r['icon_large']) . '" icon_no="' . xmlentities($absolute_server_URI . $r['icon_no']) . '" icon_undef="' . xmlentities($absolute_server_URI . $r['icon_undef']) . '">' . xmlcdata($r['text_long']) . '</attr>' . "\n");
```
Yeah, but still, though not included here, we have a new xmlentities function in export.inc.php.
I prefer to keep this rename because it is a local reimplementation of an already existing library (albeit not included here) function.
